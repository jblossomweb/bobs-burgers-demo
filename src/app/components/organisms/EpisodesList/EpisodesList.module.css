.list {
  @apply w-full
    flex flex-wrap flex-row
    items-center justify-center
    m-4 gap-4
    animate-fadein;
}
