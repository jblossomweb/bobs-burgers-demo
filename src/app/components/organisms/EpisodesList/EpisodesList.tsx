import React from 'react'
import type Episode from 'app/models/Episode'
import EpisodeCard from 'app/components/molecules/EpisodeCard'
import styles from './EpisodesList.module.css'

export interface Props {
  episodes: Episode[]
}

const EpisodesList: React.FC<Props> = ({ episodes }) => (
  <div className={styles.list}>
    {episodes.map((episode) => (
      <EpisodeCard key={episode.id} episode={episode} />
    ))}
  </div>
)

export default EpisodesList
