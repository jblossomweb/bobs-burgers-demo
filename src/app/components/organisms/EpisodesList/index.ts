export { default, type Props } from './EpisodesList'
