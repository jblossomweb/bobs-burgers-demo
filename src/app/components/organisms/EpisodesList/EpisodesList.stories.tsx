import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'
import mockEpisodes from 'app/models/__mocks__/episodes.json'

import EpisodesList, { type Props } from './EpisodesList'

export default {
  title: 'components/organisms/EpisodesList',
  component: EpisodesList,
  argTypes: {
    episodes: { control: 'object' },
  },
} as ComponentMeta<typeof EpisodesList>

const Template: ComponentStory<typeof EpisodesList> = ({ episodes }: Props) => (
  <EpisodesList episodes={episodes} />
)

export const Sample = Template.bind({})
Sample.args = {
  episodes: mockEpisodes as Props['episodes'],
}
