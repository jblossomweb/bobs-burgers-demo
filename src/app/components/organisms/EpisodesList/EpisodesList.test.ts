import storySnaps from 'test/utils/storySnaps'
import * as stories from './EpisodesList.stories'

describe('app/components/organisms/EpisodesList', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
