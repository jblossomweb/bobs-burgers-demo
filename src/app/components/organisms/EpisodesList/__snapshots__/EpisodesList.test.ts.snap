// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/organisms/EpisodesList stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="list"
  >
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Human Flesh" (a.k.a. "Pilot")
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              January 9, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/1"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Crawl Space"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              January 16, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/2"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Sacred Cow"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              January 23, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/3"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Sexy Dance Fighting"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              February 13, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/4"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Hamburger Dinner Theater"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              February 20, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/5"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Sheesh! Cab, Bob?"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              March 6, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/6"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Bed & Breakfast"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              March 13, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/7"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Art Crawl"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              March 20, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/8"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Spaghetti Western and Meatballs"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              March 27, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/9"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Burger War"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              April 10, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/10"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Weekend at Mort's"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              May 8, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/11"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Lobsterfest"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              May 15, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/12"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "Torpedo"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              May 22, 2011
            </p>
          </div>
        </div>
        <a
          href="/episodes/13"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
    <div
      class="container"
      style="width: 400px;"
    >
      <div
        class="card card"
      >
        <div
          class="body"
        >
          <h2
            class="title"
          >
            "The Belchies"
          </h2>
          <div
            class="content"
          >
            <p
              class="info"
            >
              March 11, 2012
            </p>
          </div>
        </div>
        <a
          href="/episodes/14"
        >
          <button
            class="button button"
            type="button"
          >
            Details...
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
